lib/solr/connection.rb

```diff
@@ -1,38 +1,37 @@
 module Solr
   # low-level connection that can do network requests to Solr
   class Connection
-
     def initialize(url, faraday_options: {})
-      @raw_connection = Faraday.new(url, faraday_options)
+      # Allow mock the connection for testing
+      @raw_connection = Solr.connection || Faraday.new(url, faraday_options)
```

Comment:
I would prefer if you are more explicit. For example.

```
require 'solrb/testing'

Solrb.test_connection = double(...)

# or another approach:

Solrb.configure do |config|
   config.test.connection = double(...)
end

That way it's very explicit what's happening.

Then inside the Solr::Connection it's very clear that you use test connection if it's available.
```

W/out the require this attribute shouldn't exist.